### PR TITLE
[Process RFC] Clarify Community Strategy Decision Process

### PR DESCRIPTION
**Pull request by tqchen**
- [rendered](https://github.com/tqchen/tvm-rfcs/blob/strategy/rfcs/0102-clarify-strategy-decision-process.md)

### CODE REVIEW

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Machine Learning Compilation (MLC) is an emerging field in fast development.
+With the tremendous help from the whole community, it’s exciting to see that TVM delivers significant needs from and to
+developers and thus has become widely popular in both academia and industry.
+
+As the community pushes for different goals that help each other, naturally, there
+are strategy decision points about overall directions and new modules adoptions.
+These decisions are not fine-grained code-level changes but are important for a
+community to be viable in the long term.
+The process of bringing those changes is less clarified to the community, and hurdles can be high.
+We have made attempts in the past to bring more verbose processes in a less successful.
```
Comment (csullivan) [inline rfcs/0102-clarify-strategy-decision-process.md:19]:
```suggestion
We have made attempts in the past to bring a more verbose processes, but this has proven to be less successful.
```

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+leading to stagnation and lack of participation from volunteer members.
+
+We are in a different world now in the case of ML/AI ecosystem, and it is critical for
+the community to be able to make collective decisions together and empower the community.
+Following the practices of existing ASF projects (e.g. hadoop), we propose to use a simple process for strategic decisions.
+
+## Proposal: Strategy Decision Process
+
+We propose the following clarification of the strategy decision process:
+It takes Lazy 2/3 majority of binding decisions to make strategic decisions in the TVM community, including:
```
Comment (junrushao) [inline rfcs/0102-clarify-strategy-decision-process.md:31]: I think it’s called “super majority” technically

```suggestion
It takes lazy 2/3 super majority of binding decisions to make strategic decisions in the TVM community, including:
```

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+- Adoption of a guidance-level community strategy to enable new directions or overall project evolution.
+- Establishment of a new module in the project.
+- Adoption of a new codebase: When the codebase for an existing, released product is to be replaced with an alternative codebase.
+  If such a vote fails to gain approval, the existing code base will continue. This also covers the creation of new sub-projects within the project.
+
+All these decisions are made after community conversations that get captured as part of the summary.
```
Comment (junrushao) [inline rfcs/0102-clarify-strategy-decision-process.md:38]: Is there a guideline on a hard limit on how long the discussion should continue before voting? Forcing a vote after sufficient discussion should be a legit option to come to a conclusion

Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:38]: I think a rule of thumb is no new information in terms of opinion being presented, But again we want to be simple here and leave that to the PMC and the community

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 2 above)
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:31]: I used the terminology here https://hadoop.apache.org/bylaws.html lazy 2/3 majority

Comment (csullivan) [inline rfcs/0102-clarify-strategy-decision-process.md]: I had to look up Lazy majority and found Apache Hive has a nice table describing various forms of reaching approval. I appreciate TVM's consideration of incorporating both Lazy majority and 2/3 majority. By Lazy majority are we defining it as,



>Lazy Majority | A lazy majority vote requires 3 binding +1 votes and more binding +1 votes that -1 votes.


Perhaps it would be good to include that definition.

**Review by csullivan — COMMENTED** [review]
I'm in favor of this change.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 2 above)
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md]: Great pt, added to the description

**Comment by Johnson9009** [comment]
+1

**Comment by junrushao** [comment]
I'm in favor of this. A healthy community needs progress and keep updated to the latest trend, and a super majority vote is for now the best way we could think of to balance the need for stability while being against stagnation

**Comment by comaniac** [comment]
+1

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+With the tremendous help from the whole community, it’s exciting to see that TVM delivers significant needs from and to
+developers and thus has become widely popular in both academia and industry.
+
+As the community pushes for different goals that help each other, naturally, there
+are strategy decision points about overall directions and new modules adoptions.
+These decisions are not fine-grained code-level changes but are important for a
+community to be viable in the long term.
+The process of bringing those changes is less clarified to the community, and hurdles can be high.
+We have made attempts in the past to bring a more verbose processes, but this has proven to be less successful.
+One observation is that it is hard for broader volunteer developers and community members to follow complicated processes.
```
Comment (leandron) [inline rfcs/0102-clarify-strategy-decision-process.md:20]: Can you clarify which evidence we have to support this statement?

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+## Proposal: Strategy Decision Process
+
+We propose the following clarification of the strategy decision process:
+It takes lazy 2/3 majority (at least 3 votes and twice as many +1 votes as -1 votes)
+of binding decisions to make strategic decisions in the TVM community, including:
+
+- Adoption of a guidance-level community strategy to enable new directions or overall project evolution.
+- Establishment of a new module in the project.
+- Adoption of a new codebase: When the codebase for an existing, released product is to be replaced with an alternative codebase.
+  If such a vote fails to gain approval, the existing code base will continue. This also covers the creation of new sub-projects within the project.
```
Comment (leandron) [inline rfcs/0102-clarify-strategy-decision-process.md:37]: It would be good, as we will be eventually voting on this RFC, that the proposal becomes more specific on what it is talking about. The current text is a bit vague on what the 2/3 votes majority applies to.

I'm particularly concerned about the terminology "including:", as it seems too open ended. I suggest coming up with a specific list of well defined items that 2/3 votes apply.

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Following the practices of existing ASF projects (e.g. hadoop), we propose to use a simple process for strategic decisions.
+
+## Proposal: Strategy Decision Process
+
+We propose the following clarification of the strategy decision process:
+It takes lazy 2/3 majority (at least 3 votes and twice as many +1 votes as -1 votes)
+of binding decisions to make strategic decisions in the TVM community, including:
+
+- Adoption of a guidance-level community strategy to enable new directions or overall project evolution.
+- Establishment of a new module in the project.
```
Comment (leandron) [inline rfcs/0102-clarify-strategy-decision-process.md:35]: Establishment of new modules in the project is the main subject of the very detailed (ongoing) RFC https://github.com/apache/tvm-rfcs/pull/95, so I'm wondering what's the strategy here.

Do we want to close that RFC as rejected and expand this one, or remove this point from here and continue the discussion on https://github.com/apache/tvm-rfcs/pull/95?

**Review by leandron — CHANGES_REQUESTED** [review]
The topic of decision making in the TVM community is a long standing topic, present in the ongoing RFC https://github.com/apache/tvm-rfcs/pull/95, https://github.com/apache/tvm-rfcs/pull/91 and https://github.com/apache/tvm-rfcs/pull/89 as well as numerous posts in the forum, in which many member of the community expresses their concerns with the proposals.

As with the previous proposals, this one needs to be improved on a few points for actual clarification, so that everybody understands what is being voted and the particular consequences it will have for their use-cases. We shouldn't rush into it IMHO.

I understand the fact a subset of the community wants to bring changes faster into the TVM stack for the benefit of "a field in fast development" or to attract more contributors, etc.

On the flip side, many see value in having a stable architecture with a set of components we can evolve long term in a coherent way, that would require us to be mindful when bringing new alternatives to solutions we already have on the stack. It might become too costly for some to keep up with gigantic puzzle of features that are just too complicated to make work.

All that is to say that fundamentally changing the decision making to make the barrier lower for key elements like "new direction or overall project evolution" is a new direction to the project in itself.

I'm not in favour of this RFC as is, but eventually it is for us collectively to decide and think what's the impact for our individual use cases.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 5 above)
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:37]: This is the same language copied from existing ASF communities, e.g. hadoop

Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:37]: Sorry to disagree but this is not the same as Hadoop (at least not the one I'm seeing at https://hadoop.apache.org/bylaws.html), given they have a specific list of topics including which type of decision making applies to each of them.

The text I'm reading in the proposed RFC rather open ended and can be used to justify any change.

Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:37]: this particular item comes from part of hadoop note

```
Adoption of New Codebase

When the codebase for an existing, released product is to be replaced with an alternative codebase. If such a vote fails to gain approval, the existing code base will continue.

This also covers the creation of new sub-projects within the project

Lazy 2/3 majority of PMC members
```

Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md]: I removed the terminology including, and points to the specific itemized list

**Comment by junrushao** [comment]
I was a bit confused, and would love to clarify my points first, and ask @leandron to clarify some of your points :-)

> It might become too costly for some to keep up with gigantic puzzle of features that are just too complicated to make work.

To clarify, according to my read, this RFC is particularly designed to be **extremely conservative** that a decision making should get **super majority (2/3)** of the votes to pass. It means the will of a single person or identity is insufficient to push forward anything unless they convince most of the community members.

Intuitively, if a super majority of the community firmly believe in a progress to be made via formal voting, I would agree that this is a technical direction that the community believes in and should invest in.

Of course, upon any single commit of any code change, there will be less active contributors feeling left behind, and we do need to care about those contributors even if a super majority think the contrary. Therefore, I would propose to attach a guideline: any legacy code path, if conflicts with a new direction, it has to be preserved in a separate branch. To be clear, there is no case yet that a decision introduces incompatibility against the existing ones, but as PMC members, I should give **sufficient emotional support** to those who could not catch up.

> On the flip side, many see value in having a stable architecture with a set of components we can evolve long term in a coherent way, that would require us to be mindful when bringing new alternatives to solutions we already have on the stack

I share the same sentiment with you that I wish some of my softwares never got updated, for example, LLVM breaks upon almost every release, and Cython just breaks recently which is a huge headache. Stability is indeed an important issue to me, and therefore, if a “stable architecture” already copes with new technologies out of box, I would personally prefer not to upgrade, just to save my own time.

Meanwhile, as a PMC member, I would take responsibility to a) hear the voice of super majority, and b) make concrete progress to keep the software updated to the latest challenge, which further becomes neither my personal will, nor stability concern wishing for zero change or zero break, but a grave concern on a) community stagnation; b) being out of touch from major usecases.

> I understand the fact a subset of the community wants to bring changes faster into the TVM stack for the benefit of "a field in fast development" or to attract more contributors, etc.

I apologize in advance if I misread as a non-native speaker, but correct me if I was wrong, as a PMC member just like you, I was slightly surprised about your wording on this particular scenario. Given the context that this is an extremely conservative proposal that requires super majority to push forward any technical decision, I would love to have you re-examine the following assumption:
- A super majority is described as "a subset of the community". Do you want to clarify if it's you believe super majority is representative in your reasoning?
- The motivation of having new features, as you described, is to "attract more contributors" instead of merely coping with latest needs in the field. Do you want to clarify if you believe there is any need to cope with latest needs?

To summarize, @leandron, do you agree with me that we value community over code? I'm really grateful to have such a helpful and collaborative super majority of the community, and we concrete deliver for them all the time, making our solution as fast, robust and easy-to-use, solving community problems at large scale.

**Comment by Hzfengsy** [comment]
+1

**Comment by leandron** [comment]
> The RFC is about how we operate as a community, and it's not necessarily related to "gigantic puzzle of features". To clarify, according to my read, this RFC is particularly designed to be extremely conservative that a decision making should get super majority (2/3) of the votes to pass. It means the will of a single person or identity is insufficient to push forward anything unless they convince most of the community members.
>
> As a community of diverse needs, of course, we have different opinions on how we could drive our technical needs, which is a process issue of any community operation. If a super majority of the community firmly believe in one approach of how, I would agree and support it no matter what I personally believe in. The RFC is just about that the community needs a way to drive technical resolution.

I wasn't aware of these motivations, I think they should be stated in the RFC text.

> I share the same sentiment with you that I wish some of my softwares never got updated, for example, LLVM breaks upon almost every release, and Cython just breaks recently which is a huge headache (and meanwhile, they constantly bring in coolest new features!). Stability is indeed an important issue to me, and therefore, if a “stable architecture” already copes with new challenges out of box, I would personally prefer not to upgrade, just to save my own time.

We are getting confused with the difference between a stable architecture (components that interact with each other), with API, which is the specific function calls and classes design to make something work. What I'm arguing is that we should prioritise evolving our existing components e.g. IRs, rather than come up with something new every couple years.

Cython and LLVM are great software with mature communities, being used in many production cases and - I'd argue - very stable.

This is probably one of the reason their contribution guides are clear w.r.t changes, for example Cython mention the development larger features in branches before they get into the main codebase ([source](https://github.com/cython/cython/wiki/HackerGuide#feature-branches)) and LLVM has this note in their development guide: **"we need to strike a balance between being inclusive of new ideas and people and the cost of ongoing maintenance that new code requires. As such, we have a general support policy for introducing major new components into the LLVM world, depending on the degree of detail and responsibility required. Core projects need a higher degree of scrutiny than peripheral projects, and the latter may have additional differences."**

So, if said projects keeping control of their architectures break sometimes, imagine for project that are less careful with that, how would that be?

> Meanwhile, as a PMC member, I would take **PMC responsibility** to a) hear the voice of super majority, and b) make concrete progress to keep the software updated to the latest challenge, which further becomes neither my personal will, nor stability concern wishing for zero change or zero break, but grave concern accordingly on a) community stagnation; b) being out of touch from major usecases.

I this this paragraph misinterpret the role of the PMC, which is not to "hear the voice of super majority". It says, instead, among other things: "to further the long-term development and health of the community as a whole, and to ensure that balanced and wide scale peer review and collaboration takes place." ([source](https://www.apache.org/foundation/how-it-works/#pmc)). 

I think it is not welcoming and correct to say that there is this "super majority" that crushes everyone else's opinion. At least this is not the way I think it should be.

> > I understand the fact a subset of the community wants to bring changes faster into the TVM stack for the benefit of "a field in fast development" or to attract more contributors, etc.
> 
> I apologize in advance if I misread as a non-native speaker, but correct me if I was wrong, as a PMC member just like you, I was slightly surprised about your wording on this particular scenario. Given the context that this is an extremely conservative proposal that requires super majority to push forward any technical decision, I would love to have you re-examine the following assumption:
> 
> * A super majority is described as "a subset of the community". Do you want to clarify if it's you believe super majority is representative in your reasoning?

A "subset" is exactly what it is.

> * The motivation of having new features, as you described, is to "attract more contributors" instead of merely coping with latest needs in the field. Do you want to clarify if you believe there is any need to cope with latest needs?

Sorry, I'm not aware of statistics in this area. I'm just representing a view I took from previous numerous discussions around this topic, that I referred in my previous post.

> To summarize, @leandron, do you agree with me that we value community over code? I'm really grateful to have such a helpful and collaborative super majority of the community, and we concrete deliver for them all the time, making our solution as fast, robust and easy-to-use, solving community problems at large scale.

This text is already off-topic by a lot, so I suggest rather than posting passive-aggressive questions such as whether long term contributors of this project agree with "community over code", we focus on clarifying the RFC text, so that the community understands what is it that we're changing once this goes to vote.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 6 above)
Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:35]: Sorry, this response is not addressing the question.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 5 above)
Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md]: Thanks

**Comment by tqchen** [comment]
It is clear from current conversations and past conversations that there are different opinions on how we should operate as a community.

These include what we should prioritize (e.g., “prioritize evolving our existing components, e.g., IR”), how we evolve core components, and how to “ensure long-term development and health of the community as a whole.”

It is also clear there are different interpretations of the same text; for example, some opinions about what approach qualifies as “ensure long-term development” or “evolving existing core components.”  

This proposal comes with the observation that there is no single, dogmatic way or interpretation to approach all situations. There are many ways a project can be successful. There should be holistic consideration from a technical perspective, the current state of the AI/ML ecosystem, and the demand from the community.  This RFC do not suggest a specific path, or interpretation of the how.

Unfortunately, some of the hows, or interpretation of hows, are exclusive to each other when it comes down to the operational process. By stagnating, we are implicitly taking a path that is favored by some, but dismiss the opinions of others, sometimes the majority of the community.

It shall be the decision of the community collectively to choose the how(and path) to take after conversations.  That means each one of us should do our best to convey our idea of how (technical strategies with concrete technical reasonings, approach of supporting emerging needs, approach to maintain existing code) to the community. The community collectively chooses.

We would anticipate everyone to put the factors (such as long-term development, evolving our existing components, community growth, the need of stability on some modules) into consideration. Of course everyone would have their own interpretation of how these goals boils down to a concrete decision.

In short, this is a question of process for the community to collectively make decisions on how (and implicitly their interpretation to some extent), and that is what this RFC is about.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 6 above)
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:35]: I think it is up to @Hzfengsy to decide what to do with the other RFC.   (addressing your question on what to do with existing RFC).  

I would remark that this reply is not taking the either the options that you listed, which contains specific operational ways of how. Instead it addresses the two action items:

- The other RFC: leave it up to @Hzfengsy , however i don't think any action needs to be taken until we conclude either one of the two.
- This RFC, I do not intend to expand on the particular item, as being stated, the goal is to is to come up with simple, less verbose process that everyone understand, as verbatim it is quite clear I think to many community members.

Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:35]: I don't think the community benefits of intentionally having unwritten "less verbose" processes, but I understand from the message above that there is no intention of making the current text more clear, which is part defats the purpose of having an RFC process at all.

Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:35]: Thank you for your inputs. The purpose of the RFC is indeed to collect the inputs, that includes yours as well as broader set of community members.

I understand that you do not agree the simple text is good, as you have made your point very clearly -- this comes down to disagreement on how to interpret clarity. 

Unfortunately one can not both expand the text(to fulfill one request) while also keep the text simple (to fulfill the need of others). After considering your input, as well as inputs from broader community members, I think it is good course of action to keep the proposal text as it is after looking at the inputs so far and let the community decide whether to accept it.

At any time point, you are more than welcomed to bring up a proposal to the community  as well that is independent from this one.

**Comment by junrushao** [comment]
As a diverse community, there are clearly different opinions on rules and guidelines of community operation, and it's not unusually that people agree to disagree. As PMC members, it's our responsibility to take actions to navigate the way the community operates, hearing voices from the community, keeping our technical decisions up-to-date to the latest challenges in industry, and that is exactly why we need this process RFC - Clarify the rules and guidelines of community operation.

> I think this paragraph misinterpret the role of the PMC, which is not to "hear the voice of super majority". It says, instead, among other things: "to further the long-term development and health of the community as a whole, and to ensure that balanced and wide scale peer review and collaboration takes place." ([source](https://www.apache.org/foundation/how-it-works/#pmc)).

Textual interpretation, just as @tqchen said, is usually subject to personal opinions. This is why we need a process RFC to clarify how decision making process could potentially incorporate community needs. This is how I interpret "community as a whole".

> I think it is not welcoming and correct to say that there is this "super majority" that crushes everyone else's opinion. At least this is not the way I think it should be.

I see what you mean. Thanks for clarifying your opinion as a PMC member.

**Comment by yongwww** [comment]
+1! I really like this proposal! The ability to move fast is crucial in adapting to the ever-evolving AI infra. I like the concept of lazy majority of binding decisions, it could facilitate the decision-making process and make sense to me.

**Comment by csullivan** [comment]
Hi @leandron! Would you kindly consider posting the text changes you would like to see in order for you to lift your changes requested? This may help keep discussions concise as well as help others in the community understand your desires most directly. 

If convenient to you, it could help discussion for these to be submitted as code changes suggestions in the github review. Reading from the document changes your propose would be helpful to avoid inflation of discussion on irrelevant topics that can frequently happen in committee and discussion threads. 

Thanks for considering

**Review by leandron — CHANGES_REQUESTED** [review]
> Hi @leandron! Would you kindly consider posting the text changes you would like to see in order for you to lift your changes requested? This may help keep discussions concise as well as help others in the community understand your desires most directly.
> 
> If convenient to you, it could help discussion for these to be submitted as code changes suggestions in the github review. Reading from the document changes you propose would be helpful to avoid inflation of discussion on irrelevant topics that can frequently happen in committee and discussion threads.
> 
> Thanks for considering

Thanks @csullivan - here it is.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 4 above)
Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:20]: Closing this as it is covered in the last review as code changes.

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+Machine Learning Compilation (MLC) is an emerging field in fast development.
+With the tremendous help from the whole community, it’s exciting to see that TVM delivers significant needs from and to
+developers and thus has become widely popular in both academia and industry.
+
+As the community pushes for different goals that help each other, naturally, there
+are strategy decision points about overall directions and new modules adoptions.
+These decisions are not fine-grained code-level changes but are important for a
+community to be viable in the long term.
+The process of bringing those changes is less clarified to the community, and hurdles can be high.
+We have made attempts in the past to bring a more verbose processes, but this has proven to be less successful.
```
[the remark this answers is not in the document]
Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:19]: Closing this as it is covered in the last review as code changes.

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+Authors: @tqchen
+
+- Feature Name: [Process RFC] Clarify Community Strategy Decision Process
```
[the remark this answers is not in the document]
Comment (leandron) [reply rfcs/0102-clarify-strategy-decision-process.md:3]: Closing this as it is covered in the last review as code changes.

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+## Proposal: Strategy Decision Process
+
+We propose the following clarification of the strategy decision process:
+It takes lazy 2/3 majority (at least 3 votes and twice as many +1 votes as -1 votes)
+of binding decisions to make the following strategic decisions in the TVM community:
+
+- Adoption of a guidance-level community strategy to enable new directions or overall project evolution.
+- Establishment of a new module in the project.
+- Adoption of a new codebase: When the codebase for an existing, released product is to be replaced with an alternative codebase.
```
[the remark this answers is not in the document]
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:36]: I do not think this refer to any of the components, but just the codebase of that the project develops on. As verbatim it is very is clear, an alternative codebase of the product.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:3]: Thank you for your input

ASF requires consensus on code-level contributions, but the case of long term strategy decisions are left  unspecified for per project.

But I appreciate that some may have a different view, and that is a process issue(e.g. how the community operate), and is being decided procedurally by each ASF community, which we are doing right now.

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  If such a vote fails to gain approval, the existing code base will continue. This also covers the creation of new sub-projects within the project.
+
+All these decisions are made after community conversations that get captured as part of the summary.
```
[the remark this answers is not in the document]
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:39]: Thank you for your input. There are two possible ways to view how we want to approach things:

-  W0: There needs to be extra RFCs as being requested in the text.
-  W1: The discussion already cover the information needed, and the decision process provide an resolution in a simple  and clear fashion that is understood by community members, in that case no RFC is necessary.

I understand that the request is W0. Given both W0 and W1 are related to how we operate, and they are exclusive to each other, after reviewing your inputs and others, I plan to keep the text in the form W1 for now and see whether community prefer the other one

File: rfcs/0102-clarify-strategy-decision-process.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+the community to be able to make collective decisions together and empower the community.
+Following the practices of existing ASF projects (e.g. hadoop), we propose to use a simple process for strategic decisions.
+
+## Proposal: Strategy Decision Process
+
+We propose the following clarification of the strategy decision process:
+It takes lazy 2/3 majority (at least 3 votes and twice as many +1 votes as -1 votes)
+of binding decisions to make the following strategic decisions in the TVM community:
+
+- Adoption of a guidance-level community strategy to enable new directions or overall project evolution.
```
[the remark this answers is not in the document]
Comment (tqchen) [reply rfcs/0102-clarify-strategy-decision-process.md:34]: I am not sure I get it. strategy do not necessarily entail deprecation or breaking changes :)

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Johnson9009) [reply rfcs/0102-clarify-strategy-decision-process.md:39]: Everybody like RFC, so do I, because we can know the details of the design, everybody hate RFC when his changes is very obvious or clear, so do I. Instead of a beautiful RFC and slow implement, I perfer a workable code.

File: rfcs/0102-clarify-strategy-decision-process.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Johnson9009) [reply rfcs/0102-clarify-strategy-decision-process.md:20]: e.g. I can't spend too much time to read a long long discussion article, if a commercial company will do a decision like this way, it will loss all customers.

**Comment by YuchenJin** [comment]
I fully support this RFC to require a 2/3 majority to make strategic decisions in TVM.

I'm in favor of the RFC text as it is now. It's clear and concise, while precise and effective.

**Comment by hsaputra** [comment]
+1 for this proposal

This will allow the project and community to move fast given the nature of the space that the project is in.
Very clear description on what the expectation from the community and the project.

NOTE: I saw a comment about having "value in having a stable architecture with a set of components", this is can be achieve via Git management and releases. We should NOT block progress for OSS projects as if they are internal enterprise based projects that need permissions on some VPs or business leaders to make decisions. 

Reminder that PMC IS the leadership of any ASF Top Level Projects. So member of PMC needs to be able to balance and switch hats as they are contributing to the success of the project.

**Comment by hsaputra** [comment]
> We are getting confused with the difference between a stable architecture (components that interact with each other), with API, which is the specific function calls and classes design to make something work. What I'm arguing is that we should prioritise evolving our existing components e.g. IRs, rather than come up with something new every couple years.
> 
> Cython and LLVM are great software with mature communities, being used in many production cases and - I'd argue - very stable.

I do Not think "we" are confused. Coming up with something new every couple of years actually should be happening for open source projects. API stability is another dimensions of evolution and progression that we could manage via versioning and release management.

I would not recommend to compare Apache TVM, or ASF, with other OSS projects and ecosystem. They are different and we will never try to operate and follow other communities on how to operate.

**Comment by psrivas2** [comment]
+1
I support this proposal. We need to be agile to be competitive in this fast moving space of AI / ML ecosystem, while at the same time ensure we are building robust, stable software architecture. 2/3rd majority strikes the right balance between those two options.

**Comment by tqchen** [comment]
Thank you everyone for your inputs so far. We have had many related conversations over the past year and get collective inputs on different views to approach this process. Based on the inputs so far over the past year, I opened a vote to bring forward the version that most in the discussion thread prefers so we can bring clarity to this process.

https://github.com/apache/tvm/issues/15521

Thank you

**Comment by mbaret** [comment]
+1
I think this is a substantially improved proposal compared with the similar RFC from last year, being more concise and using a 2/3rds majority which feels appropriate.

**Comment by sequoiar** [comment]
+1

**Review by Hzfengsy — APPROVED** [review]

**Comment by tqchen** [comment]
Link to the voting result thread 

https://github.com/apache/tvm/issues/15618

https://lists.apache.org/thread/bd0ph9j98r6sjm3wtp9174zgqqfhskt6
